src/Renderer.php

```diff
@@ -206,6 +209,91 @@ public function inlineJson($values)
         return htmlspecialchars(json_encode($values), ENT_QUOTES, 'UTF-8');
     }
 
+    /**
+     * Given a break stripe's content and index within the page, returns an array of list items for rendering in a table of contents
+     *
+     * @param  string $content the html in the textblock stripe as a string
+     * @param  int    $index   the index of the stripe within the page
+     * @return array
+     *
+     */
+    private function getBreakStripeTableItems($content, $index) {
+        return [[
+            'level' => 1,
+            'text' => strip_tags($content),
+            'id' => 'vssl-stripe--break--heading-' . $index
+        ]];
+    }
+
+    /**
+     * Given a textblock stripe's content and index within the page, returns an array of list items for rendering in a table of contents
+     *
+     * @param  string $content the html in the textblock stripe as a string
+     * @param  int    $index   the index of the stripe within the page
+     * @return array
+     *
+     */
+    private function getTextblockStripeTableItems($content, $index) {
+        $dom = new DOMDocument();
+        $dom->loadHTML($content ?? '');
+        $xpath = new DOMXPath($dom);
+        $headers = $xpath->query('//h1 | //h2');
+
+        $listItems = [];
+        $h1Count = 0;
+        $h2Count = 0;
+        foreach ($headers as $header) {
+            $tagName = $header->tagName;
+            $prefix = 'vssl-stripe--textblock--heading-';
+            $headerIndex = $tagName === 'h1' ? $h1Count : $h2Count;
+            $id = $prefix . $index . '-' . $tagName . '-' . $headerIndex;
+
+            array_push($listItems, [
+                'level' => $tagName === 'h1' ? 1 : 2,
+                'text' => $header->nodeValue,
+                'id' => $id
+            ]);
+
+            if ($tagName === 'h1') $h1Count++;
+            else $h2Count++;
+        }
+        return $listItems;
+    }
+
+    /**
+     * Returns a table of contents based on the contents of the page
+     *
+     * @return string
+     */
+    public function tableOfContents($scope)
+    {
+        if (empty($this->data['stripes'])) return '';
+
+        $stripes = $this->data['stripes'];
+        $items = array_map(function ($stripe, $index) use ($scope) {
+            if (
+                (empty($scope) || strpos($scope, 'break') !== false) &&
```

Comment:
What are the `$scope` options? Maybe pass in a string (single scope) or an array (e.g. `'break'` or `['break', 'textblock']` or whatever), so you don't have to do a string check like this.